internal/membership_listener.go

```diff
@@ -92,6 +116,9 @@ func (c *membershipListener) Start(ctx context.Context) {
 			}
 
 			c.wp.Submit(func() {
+				ctx, cancel := context.WithCancel(ctx)
```

Comment:
why content with cancel?
You don't use the cancel.
Expect at the end of the function but at this moment, the processing is terminated.